grovedb/src/replication.rs

```diff
@@ -277,99 +67,119 @@ impl GroveDb {
-                                match chunk_res {
-                                    Ok((chunk, _)) => match util_encode_vec_ops(chunk) {
-                                        Ok(op_bytes) => Ok(op_bytes),
-                                        Err(_) => Err(Error::CorruptedData(
-                                            "Unable to create to load chunk".to_string(),
-                                        )),
-                                    },
-                                    Err(_) => Err(Error::CorruptedData(
-                                        "Unable to create to load chunk".to_string(),
-                                    )),
-                                }
-                            }
-                            Err(_) => Err(Error::CorruptedData(
-                                "Unable to create Chunk producer".to_string(),
-                            )),
-                        }
-                    }
-                    Some(t) => {
-                        let merk = self
-                            .open_transactional_merk_at_path(path.into(), t, None, grove_version)
-                            .value?;
-
-                        if merk.is_empty_tree().unwrap() {
-                            return Ok(vec![]);
-                        }
-
-                        let chunk_producer_res = ChunkProducer::new(&merk);
-                        match chunk_producer_res {
-                            Ok(mut chunk_producer) => {
-                                let chunk_res = chunk_producer.chunk(&chunk_id, grove_version);
-                                match chunk_res {
-                                    Ok((chunk, _)) => match util_encode_vec_ops(chunk) {
-                                        Ok(op_bytes) => Ok(op_bytes),
-                                        Err(_) => Err(Error::CorruptedData(
-                                            "Unable to create to load chunk".to_string(),
-                                        )),
-                                    },
-                                    Err(_) => Err(Error::CorruptedData(
-                                        "Unable to create to load chunk".to_string(),
-                                    )),
-                                }
-                            }
-                            Err(_) => Err(Error::CorruptedData(
-                                "Unable to create Chunk producer".to_string(),
-                            )),
-                        }
-                    }
+        let root_app_hash = self.root_hash(transaction, grove_version).value?;
+        let (chunk_prefix, root_key, is_sum_tree, chunk_id) =
+            util_split_global_chunk_id(global_chunk_id, &root_app_hash)?;
+
+        // TODO: Refactor this by writing fetch_chunk_inner (as only merk constructor
+        // and type are different)
+        match transaction {
```

Comment:
matching over `Option` variants is usually done via `if let Some(tx) = transaction {...} else {...}`

btw another pr #347 removes the necessity to diverge on transaction absence